### PR TITLE
use a lib instead of implementing it by hand

### PR DESCRIPTION
**Pull request by shoshber**
also:
* waits between api calls by default
* removed some stuff that didn't work/wasn't being exercised
* added enableTags
* added permissions constants

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -44,19 +45,36 @@ const pullDeleting = (prop, obj) => {
...
-    ),
-  )(bodyProp);
+const sleep = (exports.sleep = (seconds = 0.75) => {
+  const ms = seconds * 1000;
+  const start = Date.now();
+  while (Date.now() - start < ms) {}
+});
+
+// use to avoid tripping the discourse request limiter
+const sleepAsync = seconds =>
```
Comment (marcondag) [inline api.js]: Obviously this is where we'll need to talk about implementing a more sophisticated version of this, like I said in your other PR, we use this library for the dispute-tools to talk to discourse and don't want this kind of slowdown every time.

(Take the next part, or leave it, just a style nitpick)
And I know you probably just copied this from the discourse-node-api repository, but, from a stylistic perspective I'd love to see this be a function `sleeping` with the type: `Function<Promise<T>> -> Function<Promise<T>>` that just decorates the passed in function with whatever we come up with for rate-limiting. Then we can also easily add configuration flags for enabling or disabling rate limiting. It'd be something like this:

```javascript
const sleeping = (toSleepBefore, timeToSleep = defaultSleepSeconds) =>
  rateLimitingPreventionEnabled
    ? (...args) => doSleep(timeToSleep).then(() => toSleepBefore(...args))
    : toSleepBefore
```

File: models/admin/settings.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/**
+ * @param {discourseApi.DiscourseApiShuttle} api
+ */
+module.exports = api => ({
```
Comment (marcondag) [inline models/admin/settings.js]: :+1:

File: api.js [hunk 3]
```diff
@@ -44,19 +45,36 @@ const pullDeleting = (prop, obj) => {
...
+  new Promise(res => {
+    sleep(seconds);
+    res();
+  });
+
+const defaultSleepSeconds = 1;
+
+const makeBodiedRequest = function makeBodiedRequest(req, bodyProp, body, auth) {
+  return sleepAsync(defaultSleepSeconds).then(() => {
+    const string = qs.stringify(body);
```
Comment (marcondag) [inline api.js]: Still really surprised this works! Have you tried it out with the dispute-tools already?

Comment (shoshber) [reply api.js]: I had not! I just checked and it does work--but we definitely don't want to sleep before calling discourse.

File: api.js (hunk 1 above)
Comment (shoshber) [reply api.js]: Check out what I ended up doing!

**Pull request merged**